samcli/lib/utils/file_observer.py

```diff
@@ -264,15 +266,33 @@ def __init__(self, on_change: Callable) -> None:
         self._lock: Lock = threading.Lock()
 
     def _watch_images_events(self):
```

Comment:
Thought: should we add this exception handling as a decorator to the _watch_images_events func?

Reply:
Hmm, that's a good call out. This would let us apply this handling elsewhere (if we needed to for some reason) and would look nicer since we would be wrapping a function call, instead of everything in a try catch.